Apps/phgsgmail/gsgmail_connector.py

```diff
@@ -336,6 +434,75 @@ def _handle_run_query(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _handle_get_email(self, param):
+
+        # Implement the handler here, some basic code is already in
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Create the credentials with the required scope
+        scopes = ['https://www.googleapis.com/auth/gmail.readonly']
+
+        # Create a service here
+        self.save_progress("Creating GMail service object")
+
+        user_email = param['email']
+
+        ret_val, service = self._create_service(action_result, scopes, "gmail", "v1", user_email)
+
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        query_string = ""
+        if 'internet_message_id' in param:
+            query_string += " rfc822msgid:{0}".format(self._handle_py_ver_compat_for_input_str(param['internet_message_id']))
+
+        kwargs = {'q': query_string, 'userId': user_email}
+
+        try:
+            messages_resp = service.users().messages().list(**kwargs).execute()
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "Failed to get messages", self._get_error_message_from_exception(e))
+
+        messages = messages_resp.get('messages', [])
+        action_result.update_summary({'total_messages_returned': len(messages)})
+
+        for curr_message in messages:
+
+            curr_email_ar = ActionResult()
+
+            ret_val, email_details_resp = self._get_email_details(curr_email_ar, user_email, curr_message['id'], service, 'raw')
+
+            if (phantom.is_fail(ret_val)):
+                continue
+
+            raw_encoded = base64.urlsafe_b64decode(email_details_resp.pop('raw').encode('UTF8'))
+            msg = email.message_from_bytes(raw_encoded)
+
+            if msg.is_multipart():
+                ret_val = self._parse_multipart_msg(action_result, msg, email_details_resp, param.get('extract_attachments', False))
+
+                if phantom.is_fail(ret_val):
+                    return action_result.get_status()
+
+            else:
+                # not multipart
+                email_details_resp['email_headers'] = []
+                charset = msg.get_content_charset()
+                headers = self._get_email_headers_from_part(msg)
+                email_details_resp['email_headers'].append(headers)
+                try:
+                    email_details_resp['parsed_plain_body'] = msg.get_payload(decode=True).decode(encoding=charset, errors="ignore")
+                except Exception as e:
+                    self.debug_print("Unable to add email body: {}").format(e)
+
+            action_result.add_data(email_details_resp)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
```

Comment:
Could you share a link to your container? I'm curios to see how the Investigation looks like with multiple action results  😅

Reply:
i don't think it can return multiple. internet_message_id is a required param and should be unique (at least across a single inbox, which is also required)